crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1603,6 +1604,61 @@ impl<const N: usize> Polygon<N> {
     }
 }
 
+/// A convex polygon with `N` vertices.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Serialize, Deserialize)
+)]
+pub struct ConvexPolygon<const N: usize> {
+    /// The vertices of the `ConvexPolygon`.
+    #[cfg_attr(feature = "serialize", serde(with = "super::serde::array"))]
+    pub vertices: [Vec2; N],
+}
+impl<const N: usize> Primitive2d for ConvexPolygon<N> {}
+
+/// An error that happens when creating a `ConvexPolygon`.
+#[derive(Error, Debug, Clone)]
+pub enum ConvexPolygonError {
+    /// The created polygon is not convex.
+    #[error("The created polygon is not convex")]
+    NotConvex,
+}
+
+impl<const N: usize> ConvexPolygon<N> {
+
+    fn compute_cross_product_for(&self, a_index: usize, b_index: usize, c_index: usize) -> f32 {
+        let a = self.vertices[a_index];
+        let b = self.vertices[b_index];
+        let c = self.vertices[c_index];
+        
+        (b.x - a.x) * (c.y - a.y) - (b.y - a.y) * (c.x - a.x)
+    }
+
+    /// Create a `ConvexPolygon` from its vertices.
+    /// The convexity of the polygon is checked.
```

Comment:
I feel like the docs could be a bit better here, maybe something like:

```suggestion
    ///
    /// # Errors
    ///
    /// Returns a [`ConvexPolygonError::NotConvex`] if the `vertices` do not form a convex polygon.
```